common/core/desktop/tests/unit/kmnkbd/test_kmx_xstring.cpp

```diff
@@ -16,7 +16,635 @@
 using namespace km::kbp::kmx;
 using namespace std;
 
-void test_incxstr() {
+PKMX_WCHAR find_ptr_to_last_character(PKMX_WCHAR p_first) {
+  int length = wcslen((const wchar_t *)p_first); 
```

Comment:
Turns out a hard cast hides all manner of evil! On macOS and Linux, `wchar_t` is a UTF-8 type. Which causes great havoc because we have a UTF-16 string 😢 

```suggestion
  int length = std::u16string(p_first).length();
```

Using `u16string` is the appropriate fix.